xapian-letor/ranker/ranker.cc

```diff
@@ -354,9 +354,6 @@ Ranker::labelcomparer(const FeatureVector & firstfv, const FeatureVector& second
 std::vector<Xapian::docid>
 Ranker::rank(const Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
 {
-    if (mset.empty()) {
-	throw Xapian::InvalidArgumentError("Empty MSet. No documents could be retrieved with the given Query.\n");
-    }
```

Comment:
This is a public API so passing an empty `Xapian::MSet` here needs to behave appropriately (not crash or otherwise invoke undefined behaviour, and end up returning a valid answer - in this case, that means returning an empty vector).

If I trace the execution, we call `flist.create_feature_vectors(mset, ...)` - that takes us to `FeatureList::create_feature_vectors()` in `xapian-letor/api/featurelist.cc` which creates an empty fvec, then loops over each entry in the `MSet` (which is a no-op when the `MSet` is empty), then calls `normalise(fvec)`.

If I look at `FeatureList::normalise()` then the first line of code is:

```
    int num_features = fvec[0].get_fcount();
```

But `fvec` is still empty, so `fvec[0]` is undefined behaviour.